### ISSUE TITLE
"Select [non-]parity EPs" button for algorithm trainer

### ISSUE THREAD

**Issue by bugybunny** [sp3ctum/squanmate#36]
A button to select all parity or all non-parity EPs for the algorithm trainer additionally to `Select all` `Select none` would be nice.

**Comment by bugybunny** [comment]
Whoops, sorry didn’t look for any other open issue.

I just realized, that this does not solve the problem that I had in mind. I thought, when using CSP you never encounter some situations that have parity and only selecting non-parity EPs would solve that. But you can still have O/O when using CSP which has no parity for the whole puzzle. Do you have an idea on how to solve that? Maybe another checkbox for _only non-parity cases concerning the whole puzzle_ checkbox, where you then first choose a random EP (or maybe even one that the user has selected) for top and then based on the parity of the top EP the bottom EP. If top has parity, then select also a parity EP for bottom and vice versa.

That’s the first thing that came to my mind but I don’t like it. Unfortunately, I cannot think of a good UI solution for this problem.

**Comment by sp3ctum** [comment]
Oh that's a good point actually. I use the Lin method and only know the adjacent parity alg, and I've never permuted both layers like that. But I can see that it's a very common case still, and should definitely be supported.

Actually I think you must've arrived at the same conclusion as me: no point in learning parity cases because CSP gets rid of them. Except for Vandenberg, unlike Lin, the trainer still makes sense since there are, uh, many cases.

But I think that shouldn't be an issue actually. Currently each EP case concerns both layers anyway. So an O case would always be recognized as odd, and O/O as even.
Technically I could say it's only possible to determine parity for both layers, not just one.

But for EP cases (actually both algsets and possible future ones) I think I would like two buttons: select all odd parity cases / even cases. 

If I'm right, it should solve every use case.

**Comment by bugybunny** [comment]
Oh yeah, you’re right. I’ve made a mistake since you list all possible combinations and the user does not select single EPLLs that are randomly gonna pop up on top or bottom but rather there where the checkbox says. So you actually select the whole EP case you’re gonna get and not the separate EPLLs. I’m gonna stop writing know as this may confuse you even more :D 

Tl;dr you’re right and I made a mistake.

**Comment by sp3ctum** [comment]
Released in version 5.10 just now.

Although I have another idea to add... hold on.
